src/context/TodoProvider.js

```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext, useState } from "react";
+
+const LOCAL_STORAGE_KEY = "todosContext";
+
+const TodoContext = createContext({
+  todos: [],
+  addNewTodo: () => null,
+  deleteTodo: () => null,
+  achieveTodo: () => null,
+  updateTask: () => null,
+});
+
+export const useTodo = () => useContext(TodoContext);
+
+const TodoProvider = ({ children }) => {
+  const [todos, setTodos] = useState(
+    JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY)) || []
+  );
+
+  const addNewTodo = (todo) => {
+    const newTodos = [...todos, todo];
+    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(todo));
```

Comment:
you can also create another local function in this context which can handle saving the item in the local storage to avoid calling this logic everytime.

This would reduce the breaking points in case your saving logic has some flaws